olp-cpp-sdk-core/src/cache/DefaultCache.cpp

```diff
@@ -49,6 +51,21 @@ time_t GetRemainingExpiryTime(const std::string& key,
   return expiry;
 }
 
+/// Return success flag, cache data and remaing expiry time
+std::tuple<bool, DataPtr, time_t> GetFromCache(
+    const std::string& key, olp::cache::DiskCache& disk_cache) {
+  auto expiry = GetRemainingExpiryTime(key, disk_cache);
+  if (expiry > 0) {
```

Comment:
I would not count that we have expiry in the protected cache at all. With this we will not be able to find any key in the cache. Please change. Expiry should be optional here.